DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/parser/AnalysisCommandExternalParser.scala

```diff
@@ -0,0 +1,23 @@
+package org.opalj.support.parser
+
+import org.opalj.Commandline_base.commandlines.OpalCommandExternalParser
+import org.opalj.br.fpcf.FPCFLazyAnalysisScheduler
+import org.opalj.br.fpcf.analyses.LazyL0PurityAnalysis
+import org.opalj.tac.fpcf.analyses.purity.{LazyL1PurityAnalysis, LazyL2PurityAnalysis}
+
+/**
+ * `AnalysisCommandExternalParser` interprets a command-line argument specifying a purity analysis level
+ * and returns the corresponding `FPCFLazyAnalysisScheduler`.
+ */
+object AnalysisCommandExternalParser extends OpalCommandExternalParser[FPCFLazyAnalysisScheduler]{
```

Comment:
There is a command called AnalysisCommand which is saved in Common project. This ExternalParser should be used to parse the command while being used in the DEVELOPING_OPAL/Tools project, so I didn't write it as an inner class for purity.

Furthermore, every external parser will be named with the pattern <command name>ExternalParser. That's why this class is called AnalysisCommandExternalParser.

Reply:
I understand that. But the way it is built now, it does not support anything else but Purity, even looking at the `AnalysisCommand`, which specifies values "L0", "L1", and "L2", which do apply to our purity analyses, but not other analyses in general.

Reply:
This was done for future purposes, if this AnalysisCommand would be used else where in the future, we don't have to move it from Purity, and the AnalysisCommand could just be easily extended. But of course, if it's only used for Purity now, it can be an inner class of purity.

Reply:
But how extensible is it? This parser can't be extended like this, because a) it could return purity analyses where other analyses are expected and b) other analyses like FieldAssignability will also use levels like "L2". The `AnalysisCommand` also doesn't seem properly extensible to me, with the predefined levels either not containing levels other analyses would need or containing levels that other analyses wouldn't need.
I think we need to find a better solution here. Another problem that arises is that while many runners have a specific "main" analysis that used to use an argument like "-analysis", many also take arguments for support analyses. And I think there shouldn't be two different commands and parses to support some analysis like FieldAssignability once as "-analysis" and once as, say, "-fieldAssignability"